src/CSV/CsvManager.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace App\CSV;
+
+use App\Interfaces\CSV\CsvManagerInterface;
+use App\Interfaces\CSV\DataImportManagerInterface;
+use App\Interfaces\CSV\ImportResultInterface;
+use App\Repository\DonationRepository;
+use App\Repository\PersonRepository;
+use App\Repository\ProjectRepository;
+use App\Repository\RewardRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\Serializer\SerializerInterface;
+
+final class CsvManager implements CsvManagerInterface
+{
+    public function __construct(
+        private EntityManagerInterface $entityManager,
+        private SerializerInterface $serializer,
+        private PersonRepository $personRepository,
+        private DonationRepository $donationRepository,
+        private ProjectRepository $projectRepository,
+        private RewardRepository $rewardRepository,
+        private DataImportManagerInterface $dataImportManagerInterface
+    ) {
+    }
+
+    public function import(\SplFileInfo $filePath): ImportResultInterface
+    {
+        $this->importData($this->getDataFromFile($filePath->getPathname()));
+
+        $persons = $this->personRepository->findAll();
+        $donations = $this->donationRepository->findAll();
+        $rewards = $this->rewardRepository->findAll();
+        $projects = $this->projectRepository->findAll();
+
+        return new ImportResult($persons, $donations, $rewards, $projects);
+    }
+
+    private function importData(array $data): void
+    {
+        foreach ($data as $row) {
+
+            $person = $project = $reward = null;
+
```

Comment:
Before start import data , you should validate data maybe ?
Create a DTO with constraint validation, call validator to validate your DTO and after that if no assertion, you can import, if not maybe put it in an object response to know at the end of the process witch Lines have failed and why ?

If you use a DTO you should not have to use mixed type for rewardQuantity or amount and have correct type.